// protobuf code generated by AST_protobuf.tpl
syntax = "proto3";



package swdv.settings_service;

// Generic result message
message operation_result {
  bool result = 1;
}

// IFEX Namespace config
// IFEX Enum preset_type_t
enum preset_type_t {
    TYPE_SYSTEM = 0;
    TYPE_USER = 1;
    TYPE_FLEET = 2;
    TYPE_DEVICE = 3;
}

// IFEX Enum preset_scope_t
enum preset_scope_t {
    SCOPE_GLOBAL = 0;
    SCOPE_USER = 1;
    SCOPE_DEVICE = 2;
    SCOPE_GROUP = 3;
}

// IFEX Struct preset_metadata_t
message preset_metadata_t {
  string name = 1;
  string display_name = 2;
  string description = 3;
  preset_type_t type = 4;
  preset_scope_t scope = 5;
  uint64 created_timestamp = 6;
  uint64 modified_timestamp = 7;
  string created_by = 8;
  repeated string tags = 9;
}

// IFEX Struct preset_t
message preset_t {
  preset_metadata_t metadata = 1;
  string values = 2;
  uint32 version = 3;
}

// IFEX Struct service_method_t
message service_method_t {
  string service_name = 1;
  string service_version = 2;
  string namespace_name = 3;
  string method_name = 4;
}

// IFEX Struct preset_filter_t
message preset_filter_t {
  preset_type_t type = 1;
  preset_scope_t scope = 2;
  repeated string tags = 3;
  uint64 created_after = 4;
  string created_by = 5;
}

// IFEX Method get_presets
message get_presets_request {
  service_method_t service_method = 1;
  preset_filter_t filter = 2;
}

message get_presets_response {
  repeated preset_t presets = 1;
}

service get_presets_service {
  rpc get_presets(get_presets_request) returns (get_presets_response);
}

// IFEX Method get_preset
message get_preset_request {
  service_method_t service_method = 1;
  string preset_name = 2;
}

message get_preset_response {
  preset_t preset = 1;
  bool found = 2;
}

service get_preset_service {
  rpc get_preset(get_preset_request) returns (get_preset_response);
}

// IFEX Method create_preset
message create_preset_request {
  service_method_t service_method = 1;
  preset_t preset = 2;
}

message create_preset_response {
  bool success = 1;
  string error_message = 2;
}

service create_preset_service {
  rpc create_preset(create_preset_request) returns (create_preset_response);
}

// IFEX Method update_preset
message update_preset_request {
  service_method_t service_method = 1;
  string preset_name = 2;
  preset_t preset = 3;
}

message update_preset_response {
  bool success = 1;
  uint32 new_version = 2;
  string error_message = 3;
}

service update_preset_service {
  rpc update_preset(update_preset_request) returns (update_preset_response);
}

// IFEX Method delete_preset
message delete_preset_request {
  service_method_t service_method = 1;
  string preset_name = 2;
}

message delete_preset_response {
  bool success = 1;
  string error_message = 2;
}

service delete_preset_service {
  rpc delete_preset(delete_preset_request) returns (delete_preset_response);
}

// IFEX Method apply_preset
message apply_preset_request {
  service_method_t service_method = 1;
  string preset_name = 2;
  string merge_with = 3;
}

message apply_preset_response {
  string values = 1;
  bool success = 2;
}

service apply_preset_service {
  rpc apply_preset(apply_preset_request) returns (apply_preset_response);
}

// IFEX Method export_presets
message export_presets_request {
  service_method_t service_method = 1;
  preset_filter_t filter = 2;
}

message export_presets_response {
  string export_data = 1;
  uint32 preset_count = 2;
}

service export_presets_service {
  rpc export_presets(export_presets_request) returns (export_presets_response);
}

// IFEX Method import_presets
message import_presets_request {
  string export_data = 1;
  bool overwrite_existing = 2;
}

message import_presets_response {
  uint32 imported_count = 1;
  uint32 skipped_count = 2;
  repeated string errors = 3;
}

service import_presets_service {
  rpc import_presets(import_presets_request) returns (import_presets_response);
}

// IFEX Method validate_preset
message validate_preset_request {
  service_method_t service_method = 1;
  string ifex_schema = 2;
  string preset_values = 3;
}

message validate_preset_response {
  bool valid = 1;
  repeated string errors = 2;
}

service validate_preset_service {
  rpc validate_preset(validate_preset_request) returns (validate_preset_response);
}





